components/base/sensorcontrolled/spin.go

```diff
@@ -105,162 +52,88 @@ func (sb *sensorBase) stopSpinWithSensor(
 		timeOut = 10 * time.Second
 	}
 
-	sb.activeBackgroundWorkers.Add(1)
-	utils.ManagedGo(func() {
-		ticker := time.NewTicker(yawPollTime)
-		defer ticker.Stop()
+	for {
+		// check if we want to poll the sensor at all
+		// other API calls set this to false so that this for loop stops
 
-		for {
-			// check if we want to poll the sensor at all
-			// other API calls set this to false so that this for loop stops
-			if !sb.isPolling() {
-				ticker.Stop()
-			}
-
-			if err := ctx.Err(); err != nil {
-				ticker.Stop()
-				return
-			}
+		if !sb.isPolling() {
+			ticker.Stop()
+			sb.logger.CWarn(ctx, "Spin call interrupted by another running api")
+			return nil
+		}
 
-			select {
-			case <-ctx.Done():
-				ticker.Stop()
-				return
-			case <-ticker.C:
-				// imu readings are limited from 0 -> 360
-				currYaw, err := getCurrentYaw(sb.orientation)
-				if err != nil {
-					errCounter++
-					if errCounter > 100 {
-						sb.logger.CError(ctx, errors.Wrap(
-							err,
-							"imu sensor unreachable, 100 error counts when trying to read yaw angle, stopping base"))
-						if err = sb.Stop(ctx, nil); err != nil {
-							return
-						}
-					}
-					continue
-				}
-				errCounter = 0 // reset reading error count to zero if we are successfully reading again
+		if err := ctx.Err(); err != nil {
+			ticker.Stop()
+			return err
+		}
 
-				atTarget, overShot, minTravel := getTurnState(currYaw, startYaw, targetYaw, dir, angleDeg, errBound)
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
```

Comment:
my understanding was it would be context cancelled yeah. Should I change anything for this?

Reply:
nah, just make sure it doesn't look scary in logs or app, I try to hide context cancelled errors to not confuse users.

Reply:
kk updated to log context canceled errors, rather than returning them. This should prevent the frontend from seeing the error while still signaling that the cancels are occurring(in case the user is not doing them). Can switch the log level to warning if wanted